src/P2P/Node.hs

```diff
@@ -397,30 +397,36 @@ peerClientEnv node = peerInfoClientEnv (_p2pNodeManager node)
 -- TODO: handle paging
 --
 syncFromPeer :: P2pNode -> PeerInfo -> IO Bool
-syncFromPeer node info = runClientM sync env >>= \case
-    Left e
-        | isCertMismatch e -> do
-            logg node Warn $ "failed to sync peers from " <> showInfo info <> ": unknown certificate. Deleting peer from peer db"
-            peerDbDelete (_p2pNodePeerDb node) info
-            return False
-        | otherwise -> do
-            logg node Warn $ "failed to sync peers from " <> showInfo info <> ": " <> sshow e
-            return False
-    Right p -> do
-        caps <- getNumCapabilities
-        goods <- fmap catMaybes
-            $ traverseConcurrently (ParN $ int caps) (guardPeerDbOfNode node)
-            $ filter (\i -> me /= _peerId i)
-            $ _pageItems p
-        peerDbInsertPeerInfoList
-            (_p2pNodeNetworkId node)
-            goods
-            (_p2pNodePeerDb node)
-        return True
+syncFromPeer node info = do
+    prunePeerDb peerDb
+    runClientM sync env >>= \case
+        Left e
+            | isCertMismatch e -> do
+                logg node Warn $ "failed to sync peers from " <> showInfo info <> ": unknown certificate. Deleting peer from peer db"
+                peerDbDelete (_p2pNodePeerDb node) info
+                return False
+            | otherwise -> do
+                logg node Warn $ "failed to sync peers from " <> showInfo info <> ": " <> sshow e
+                return False
+        Right p -> do
+            peers <- peerDbSnapshot peerDb
+            goods <- fmap catMaybes
+                $ traverseConcurrently (ParN 16) (guardPeerDbOfNode node)
```

Comment:
How about a multiple of `caps`?

Reply:
I don't think it that it makes sense to bind this to caps. This is mostly about how many parallel network connections are established.